File: python /al12.py
numberToGuess = 235
count = 0
minRange = 0
maxRange = 1000

while True:
    guess = (minRange + maxRange) // 2
    count += 1

    if guess == numberToGuess:
        print(f"I guessed your number ({numberToGuess}) in {count} attempts!")
        break
    elif guess < numberToGuess:
        minRange = guess + 1
    else:
        maxRange = guess - 1

File: php/al12.php
<?php


$numberToGuess = 235;
$count = 0;
$minRange = 0;
$maxRange = 1000;



while (true) {
    $guess = floor(($minRange + $maxRange) / 2);
    $count++;

    if ($guess == $numberToGuess) {
        echo "I guessed your number ($numberToGuess) in $count attempts!\n";
        break;
    } elseif ($guess < $numberToGuess) {
        $minRange = $guess + 1;
    } else {
        $maxRange = $guess - 1;
    }
}





